kore/src/Kore/Strategies/Goal.hs

```diff
@@ -812,11 +817,13 @@ removeDestination lensRulePattern mkState goal =
         do
             removal <- removalPatterns destination configuration existentials
             when (isTop removal) (succeed . mkState $ rulePattern)
-            let configAndRemoval =
-                    fmap (configuration <*) removal
+            let configAndRemoval = fmap (configuration <*) removal
+                sideCondition =
+                    Pattern.withoutTerm configuration
+                    & SideCondition.fromCondition
             simplifiedRemoval <-
                 simplifyConditionsWithSmt
-                    SideCondition.top
+                    sideCondition
```

Comment:
Why is the side condition necessary here?

Reply:
This side condition will be used for function evaluation while simplifying the argument, before `simplifyConditionsWithSmt` sends the result to the solver.